File: 프로그래머스/0/181913. 문자열 여러 번 뒤집기/문자열 여러 번 뒤집기.py
def solution(my_string, queries):
    # 1. 문자열은 수정이 안 되므로 리스트로 변환
    arr = list(my_string)
    
    # 2. 쿼리 처리
    for s, e in queries:
        # s부터 e까지(포함)를 슬라이싱하여 뒤집은 뒤 다시 그 위치에 넣음
        arr[s : e + 1] = arr[s : e + 1][::-1]
    
    # 3. 리스트를 다시 문자열로 합쳐서 반환
    return "".join(arr)

File: 프로그래머스/0/181913. 문자열 여러 번 뒤집기/문자열 여러 번 뒤집기.cpp
#include <string>
#include <vector>

using namespace std;

string solution(string my_string, vector<vector<int>> queries) {
    string answer = "";
    int s;
    int e;
    
    for (int i = 0; i < queries.size(); i++) {
        s = queries[i][0];
        e = queries[i][1];
        
        for (int j = s; j <= (s + e) / 2; j++) {
            char tmp = my_string[j];
            my_string[j] = my_string[s + e - j];
            my_string[s + e - j] = tmp;
        }
    }
    
    answer = my_string;
    return answer;
}